src/rateEngine/RateElement.ts

```diff
@@ -9,29 +9,85 @@ import LoadProfile from './LoadProfile';
 import ValidatorFactory from './ValidatorFactory';
 import { RateComponentInterface } from './RateComponent';
 import { Error } from './validators/_Validator';
+import SurchargeAsPercent from './billingDeterminants/SurchargeAsPercent';
 
 export interface RateElementInterface {
+  id?: string;
   rateElementType: RateElementType;
-  rateComponents: Array<RateComponentInterface & BillingDeterminantFactoryInterface>;
+  rateComponents?: Array<RateComponentInterface & BillingDeterminantFactoryInterface>;
   name: string;
+  billingCategory?: BillingCategory;
+  rateElementIds?: Array<string>;
+  percent?: number;
```

Comment:
As you called out here, this doesn't seem like a generic `RateElement` concept. Is there another interface that can be created for this element that would extend `RateElement`?

Reply:
I think you are referring to internal code structure more than the user facing input/output here, so I'll speak to that.

This is a very similar pattern to the rate component / billing determinant API.

```
const rc = new RateComponent({name, charge, ...aBunchOfRandomArgsThatArePassedToBillingDeterminants})
```

This changes the the API of the RateElement like so:

```js
// from
const re = new RateElement({ name, classification, rateComponents });
// to
const re = new RateElement({ name, classification, rateComponents, ...aBunchOfRandomArgsThatCanBeUsedToGenerateRateComponents });
```

The fact that the patterns are similar makes me feel better about making the change, especially since the new requirements most definitely fit at the rate element level and not at the lower rate component level. That said, I didn't think very hard about where to put the code. I just dumped a small factory right inside the `RateElement.ts` file. The thinking there is that since this factory just covers one use case for the time being, we can punt on where to put stuff for now. In my mind, these are internal details which could be refactored. As always, suggestions and/or counterarguments are welcome!

Reply:
Maybe bundling those random rate element args into a single property would feel better? The rate component is able to put those random arguments inside the rateComponent objects....

```js
const rateElement = {
    name: 'something',
    classification: 'surcharge',
    rateComponentFactoryArgs: { // a WIP name.. would prefer something better here
      percent: 6,
   }
};  
```